service/controller/resource/verticalpodautoscaler/resource.go

```diff
@@ -182,5 +183,5 @@ func hasChanged(current, desired metav1.Object) bool {
 	c := current.(*vpa_types.VerticalPodAutoscaler)
 	d := desired.(*vpa_types.VerticalPodAutoscaler)
 
-	return !reflect.DeepEqual(c.Spec, d.Spec)
+	return !apiequality.Semantic.DeepEqual(c.Spec, d.Spec)
```

Comment:
Are there other places we need this trick? Do we need to spread it as a best practice?

Reply:
Theoretically, we should everytime we need to check equality between 2 api objects

Reply:
https://github.com/kubernetes/apimachinery/blob/17b79e78ed4ac42a54a132a541fea663fbbd80cd/pkg/api/equality/semantic.go#L27
We should advertise it as best practice but the kickoff got in the way

Reply:
See https://github.com/giantswarm/fmt/pull/134/files